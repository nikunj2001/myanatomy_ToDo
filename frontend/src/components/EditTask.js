import React,{useEffect, useState} from 'react'
import { ReactPropTypes } from 'react';
import {useDispatch, useSelector} from "react-redux";
import { editTask } from '../store/asyncMethods/taskMethods';
import { useHistory } from 'react-router-dom';
import { Toaster,toast } from 'react-hot-toast';
const EditTask = ({location}) => {
    const dispatch = useDispatch();
    const history = useHistory();
    const [state,setState] = useState({
        task: location.state.task.task,
        description: location.state.task.description,
        status: location.state.task.status
    });
    const {task} = useSelector(state=>state.TaskReducer);
    const handleChange=e=>{
        setState({...state,[e.target.name]:e.target.value});
    }
    const updateForm=e=>{
        e.preventDefault();
        if(state.task===""){
            toast.error("Task is required!");
        }
       dispatch(editTask(state,location.state.task._id));
    }
    const statusChange=e=>{
        setState({...state,status:e.currentTarget.value});
    }
    useEffect(()=>{
            if(task){
                history.push("/");
                dispatch({type:"UNSET_TASK"});
            }
    },[task]);
  return (
    <div className='form'>
        <form onSubmit={updateForm} >
                <Toaster
                    position="top-center"
                    reverseOrder={false}
                />
           <div className="form__control">
            <input  value={state.task} onChange={handleChange} type="text" name='task' placeholder='Add a Task...' />    
               </div> 
               <div className="form__control">
                 <input type="text" name='description' value={state.description} onChange={handleChange} placeholder='Description' />    
                </div>
                <div  className="form__control">
                    <select name="status" value={state.status} onChange={statusChange}>
                        <option value="pending">Pending</option>    
                        <option value="under-process">Under Process</option>    
                        <option value="completed">Completed</option>    
                    </select>  
                </div>
                 <div className="form__control">
                     <input type='submit' className='btn btn-default' value='Update Task' />
                </div>         
        </form>    
    </div>
  )
}
export default EditTask;