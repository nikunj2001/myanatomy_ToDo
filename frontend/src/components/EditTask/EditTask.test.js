import React from 'react';
import { shallow, configure } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
import EditTask from './EditTask';

Enzyme.configure({ adapter: new Adapter() })


describe("EditTask", () => {
    const location = {
        state: {
            task: {
                task: "Task-1",
                description: "Description",
                status: "pending"
            }
        }
    }
    it("should render my component", () => {
        const wrapper = shallow(<EditTask location={location} />);
        const tree = wrapper.debug();
        expect(tree).toMatchSnapshot();
    });
});