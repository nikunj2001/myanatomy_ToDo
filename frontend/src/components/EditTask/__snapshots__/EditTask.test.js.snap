// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask should render my component 1`] = `
"<div className=\\"form\\">
  <form onSubmit={[Function: updateForm]}>
    <Toaster position=\\"top-center\\" reverseOrder={false} />
    <div className=\\"form__control\\">
      <input value=\\"Task-1\\" onChange={[Function: handleChange]} type=\\"text\\" name=\\"task\\" placeholder=\\"Add a Task...\\" />
    </div>
    <div className=\\"form__control\\">
      <input type=\\"text\\" name=\\"description\\" value=\\"Description\\" onChange={[Function: handleChange]} placeholder=\\"Description\\" />
    </div>
    <div className=\\"form__control\\">
      <select name=\\"status\\" value=\\"pending\\" onChange={[Function: statusChange]}>
        <option value=\\"pending\\">
          Pending
        </option>
        <option value=\\"under-process\\">
          Under Process
        </option>
        <option value=\\"completed\\">
          Completed
        </option>
      </select>
    </div>
    <div className=\\"form__control\\">
      <input type=\\"submit\\" className=\\"btn btn-default\\" value=\\"Update Task\\" />
    </div>
  </form>
</div>"
`;

exports[`EditTask should render my component 1`] = `
"<div className=\\"form\\">
  <form onSubmit={[Function: updateForm]}>
    <Toaster position=\\"top-center\\" reverseOrder={false} />
    <div className=\\"form__control\\">
      <input value=\\"Task-1\\" onChange={[Function: handleChange]} type=\\"text\\" name=\\"task\\" placeholder=\\"Add a Task...\\" />
    </div>
    <div className=\\"form__control\\">
      <input type=\\"text\\" name=\\"description\\" value=\\"Description\\" onChange={[Function: handleChange]} placeholder=\\"Description\\" />
    </div>
    <div className=\\"form__control\\">
      <select name=\\"status\\" value=\\"pending\\" onChange={[Function: statusChange]}>
        <option value=\\"pending\\">
          Pending
        </option>
        <option value=\\"under-process\\">
          Under Process
        </option>
        <option value=\\"completed\\">
          Completed
        </option>
      </select>
    </div>
    <div className=\\"form__control\\">
      <input type=\\"submit\\" className=\\"btn btn-default\\" value=\\"Update Task\\" />
    </div>
  </form>
</div>"
`;
